docs/madr/decisions/034-default-target-ref-policy.md

```diff
@@ -0,0 +1,104 @@
+# TargetRef policies as a default
+
+* Status: accepted
+
+Technical Story: none
+
+## Context and Problem Statement
+
+At present, Kuma has two policy types:
+
+* Legacy policies with Source and Destination attributes
+* New policies based on TargetRef
+ 
+The legacy policies currently serve as the defaults. However, starting from release 2.6, the intention is to switch the default policy to TargetRef.
+
+## Considered Options
+
+1. Introduce a `policyEngine` field within the `Mesh` object, allowing the selection of the default policy version.
+
+## Decision Outcome
+
+The decision is to introduce a `policyEngine` field within the Mesh object to select the default policy version.
+
+### Positive Consequences
+
+* New deployments will start with the new default TargetRef policies.
+* Kuma remains capable of working with and applying old policies.
+* Existing meshes remain unaffected by the change.
+
+### Negative Consequences
+
+* Possibility of an additional, potentially unused field in the mesh object in the future.
+
+## Pros and Cons of the Options
+
+### Introduce a `policyEngine` field within the `Mesh` object, allowing the selection of the default policy version.
+
+
+The proposed addition to the Mesh object would appear as follows in the protobuf definition:
+
+```protobuf
+// Mesh defines configuration of a single mesh.
+message Mesh {
+
+  // ...
+    
+  enum PolicyEngine {
+    // An Undefined mode means that the policy engine wasn't defined.
+    // When creating, the default is set to TargetRef.
+    // During retrieval of old settings, distinguishing it as Legacy.
+    Undefined = 0;
+    // A Legacy mode creates old default policies.
+    Legacy = 0;
+    // A TargetRef mode creates new default targetRef policies.
+    TargetRef = 1;
+  }
+
+  // policyEngine implies which policy engine should be used as a default.
+  PolicyEngine policyEngine = 9;
+}
+```
+
+New configuration would look like:
+
+```yaml
+policyEngine: TargetRef
+```
+
+This approach enables the creation of new default target ref policies without enforcing users to only use new polcies. The control-plane will support both old and new policies, with the default `Mesh` being created using the new policies. New policies will not require `MeshTrafficPermissions`, `MeshHTTPRoute`, or `MeshTCPRoute` to enable traffic across the cluster.
```

Comment:
> New policies will not require `MeshTrafficPermissions`, `MeshHTTPRoute`, or `MeshTCPRoute` to enable traffic across the cluster.

I can understand MeshHTTPRoute/MeshTCPRoute, but why do we want to do this with MeshTrafficPermission? If there are no "permissions" in the cluster, why then the traffic should work? Also, if we have implicit `action: Allow` users are going to create denylists as opposed to allowlist. Allowlists are better for security and for the auto reachable services. 

I'm not sure if this should be discussed in this MADR, but I haven't found a relevant issue.

Reply:
Ah I see, I was thinking we want traffic by default even with strict mTLS. We should make it clear in docs that MTPs must be created before enabling the mTLS, otherwise, no requests are allowed.

Reply:
I don't think that's the case. `MeshHTTPRoute` and `MeshTCPRoute` have expectation that traffic flows by default without distinction if mTLS is enabled or not. This is the same behavior as Kubernetes. If you want to change it, you could create an appropriate `MeshTrafficPermission` policy.